main.go

```diff
@@ -2,229 +2,62 @@ package main
-			Headers    map[string][]string `json:"headers"`
-			UserAgent  string              `json:"user_agent"`
-			RequestURI string              `json:"request_uri"`
-			RemoteAddr string              `json:"remote_addr"`
-
-			// checker.Result
-			APIServerDirect string `json:"api_server_direct"`
-			APIServerDNS    string `json:"api_server_dns"`
-			MeIngress       string `json:"me_ingress"`
-			MeService       string `json:"me_service"`
-
-			// kubediscovery
-			NeighbourhoodState string                    `json:"neighbourhood_state"`
-			Neighbourhood      []kubediscovery.Neighbour `json:"neighbourhood"`
-		}
-
-		// Run checks now
-		res, haserr := chk.Run()
-		if haserr {
-			w.WriteHeader(http.StatusInternalServerError)
-		}
-
-		// Add additional data
-		out := Output{
-			APIServerDNS:       res.APIServerDNS,
-			APIServerDirect:    res.APIServerDirect,
-			MeIngress:          res.MeIngress,
-			MeService:          res.MeService,
-			Headers:            r.Header,
-			UserAgent:          r.UserAgent(),
-			RequestURI:         r.RequestURI,
-			RemoteAddr:         r.RemoteAddr,
-			Neighbourhood:      res.Neighbourhood,
-			NeighbourhoodState: res.NeighbourhoodState,
-		}
-		out.Hostname, _ = os.Hostname()
-
-		// Generate output output
-		enc := json.NewEncoder(w)
-		enc.SetIndent("", " ")
-		_ = enc.Encode(out)
-	}
-}
-
-// GenerateRoundTripper returns a custom http.RoundTripper, including the k8s
-// CA. If env KUBENURSE_INSECURE is set to true, certificates are not validated.
-func GenerateRoundTripper() (http.RoundTripper, error) {
-	// Parse environment variables
-	extraCA := os.Getenv("KUBENURSE_EXTRA_CA")
-	insecureEnv := os.Getenv("KUBENURSE_INSECURE")
-	insecure, _ := strconv.ParseBool(insecureEnv)
+		log.Println("shutting down, received signal to stop")
 
-	// Append default certpool
-	rootCAs, _ := x509.SystemCertPool()
-	if rootCAs == nil {
-		rootCAs = x509.NewCertPool()
-	}
-
-	// Append ServiceAccount cacert
-	caCert, err := ioutil.ReadFile(caFile)
-	if err != nil {
-		return nil, fmt.Errorf("could not load certificate %s: %s", caFile, err)
-	}
-
-	if ok := rootCAs.AppendCertsFromPEM(caCert); !ok {
-		return nil, errors.New("could not append ca cert to system certpool")
-	}
-
-	// Append extra CA, if set
-	if extraCA != "" {
-		//nolint:gosec
-		caCert, err := ioutil.ReadFile(extraCA)
+		// background ctx since, the "root" context is already canceled
+		shutdownCtx, shutdownCancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer shutdownCancel()
 
-		if err != nil {
-			return nil, fmt.Errorf("could not load certificate %s: %s", extraCA, err)
+		if err := server.Shutdown(shutdownCtx); err != nil {
+			log.Printf("gracefully halting kubenurse server: %s", err)
 		}
+	}()
 
-		if ok := rootCAs.AppendCertsFromPEM(caCert); !ok {
-			return nil, errors.New("could not append extra ca cert to system certpool")
-		}
-	}
+	fmt.Println(nurse) // most important line of this project
```

Comment:
really ? 😉

Reply:
This was present in the old codebase, so I think we'll leave it as "history" 😄